modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,425 @@
+		foreach ( $matches as $match ) {
+			// Filter an image match.
+			if ( isset( $images[ $match[0] ] ) ) {
+				$filtered_image = $match[0];
+				$attachment_id  = $images[ $match[0] ];
+				/**
+				 * Filters img tag that will be injected into the content.
+				 *
+				 * @param string $filtered_image the img tag with attributes being created that will
+				 *                                    replace the source img tag in the content.
+				 * @param string $context Optional. Additional context to pass to the filters.
+				 *                        Defaults to `current_filter()` when not set.
+				 * @param int $attachment_id the ID of the image attachment.
+				 *
+				 * @since 1.0.0
+				 *
+				 */
+				$filtered_image = apply_filters( 'wp_dominant_color_img_tag_add_adjust', $filtered_image, $context, $attachment_id );
+
+				if ( $filtered_image !== $match[0] ) {
+					$content = str_replace( $match[0], $filtered_image, $content );
+				}
+			}
+		}
+
+		return $content;
+	}
+
+	/**
+	 * Add checkbox setting to enable/disable blurhash for a given media.
+	 *
+	 * @param array $form_fields
+	 * @param WP_Post $post
+	 *
+	 * @return array
+	 */
+	public function add_dominant_color_media_setting( array $form_fields, WP_Post $post ) {
+		$image_meta     = wp_get_attachment_metadata( $post->ID );
+		$checked_text   = isset( $image_meta['show_dominant_color'] ) ? 'checked' : '';
+		$dominant_color = isset( $image_meta['dominant_color'] ) ? $image_meta['dominant_color'] : '';
+		$contrast_color = $this->getContrastYIQ( $dominant_color );
+		$html_input     = "<input type='checkbox' $checked_text value='1'
+			name='attachments[{$post->ID}][show_dominant_color]' id='attachments[{$post->ID}][show_dominant_color]'/>
+			<span style='background-color: #{$dominant_color}; color: {$contrast_color}; padding: 3px 6px; margin-top: 3px;vertical-align: middle;display: inline-block; '>#{$dominant_color}</span>";
+
+		$form_fields['show_dominant_color'] = array(
+			'label' => __( 'Show Dominant color as a preload placeholder', 'wp-dominant-color' ),
+			'input' => 'html',
+			'html'  => $html_input,
+		);
+
+		return $form_fields;
+	}
+
+	/**
+	 * Save dominant color setting value for a media post.
+	 *
+	 * @param array $post
+	 * @param array $attachment
+	 *
+	 * @return array
+	 */
+	public function save_dominant_color_media_setting( array $post, array $attachment ) {
+		$attachment_id = $post['ID'];
+		$image_meta    = wp_get_attachment_metadata( $attachment_id );
+
+		if ( isset( $attachment['show_dominant_color'] ) ) {
+			if ( ! isset( $image_meta['show_dominant_color'] ) ) {
+				// If enabling blurhash from media setting and image meta doesn't have any, generate a new one.
+				$image_meta                        = $this->dominant_color_metadata( $image_meta, $attachment_id );
+				$image_meta['show_dominant_color'] = $attachment['show_dominant_color'] ? '1' : '0';
```

Comment:
i might've been wrong in an above comment, sorry.. this flag only refers to it being computed?